File: php/example_code/elastictranscoder/CreateJob.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
 *  ABOUT THIS PHP SAMPLE: This sample is part of the Elastic Transcoder Developer Guide topic at
 *  https://docs.aws.amazon.com/elastictranscoder/latest/developerguide/introduction.html
 *
 */
// snippet-start:[elastictranscoder.php.create_job.complete]
// snippet-start:[elastictranscoder.php.create_job.import]
require 'vendor/autoload.php';

use Aws\ElasticTranscoder\ElasticTranscoderClient;
use Aws\Exception\AwsException;

// snippet-end:[elastictranscoder.php.create_job.import]

/**
 * Create an Elastic Transcoder job.
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

// snippet-start:[elastictranscoder.php.create_job.main]
$tmp_path = '/tmp';

// Region where you set up your AWS resources.
$region = 'us-east-2';

// Create the client for Elastic Transcoder.
$transcoder_client = new ElasticTranscoderClient([
    'profile' => 'default',
    'version' => '2012-09-25',
    'region' => $region,
    'default_caching_config' => $tmp_path,
]);

// This will generate a 480p 16:9 mp4 output.
$preset_id = '1351620000001-000020';

$pipeline_id = '1234567890112-abcdefg';
$S3_file = 'folder/filename.txt';

// All outputs will have this prefix prepended to their output key.
$output_key_prefix = 'elastic-transcoder-samples/output/';
$outputs = [
    [
        'Key' => $S3_file,
        'PresetId' => $preset_id
    ]
];

// Create the job.
try {
    $create_job_result = $transcoder_client->createJob([
        'PipelineId' => $pipeline_id,
        'Input' => ['Key' => $S3_file],
        'Outputs' => $outputs,
        'OutputKeyPrefix' => $output_key_prefix
    ]);
    var_dump($create_job_result["Job"]);
} catch (AwsException $e) {
    // output error message if fails
    echo $e->getMessage() . "\n";
}

// snippet-end:[elastictranscoder.php.create_job.main]
// snippet-end:[elastictranscoder.php.create_job.complete]

File: ruby/example_code/elastictranscoder/create_job.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# Purpose:
# Demonstrates how to create an HLS job using the AWS SDK for Ruby.

# snippet-start:[elastictranscoder.ruby.create_hls_job.import]
require "aws-sdk-elastictranscoder"
require "aws-sdk-s3"

def create_elastictranscoder_job(transcoder_client, s3_resource, pipeline_name)
  bucket_name = "transcoder-bucket-#{rand(10**4)}"
  s3_resource.create_bucket(bucket: bucket_name)

  resp = transcoder_client.create_pipeline({
                                             name: pipeline_name, # required
                                             input_bucket: bucket_name, # required
                                             output_bucket: bucket_name,
                                             role: "arn:aws:iam::260778392212:role/Elastic_Transcoder_Default_Role" # required
                                           })

  pipeline_id = resp["pipeline"]["id"]

  # This is the name of the input key that you would like to transcode.
  input_key = "Jabberwocky.mp3"

  # HLS Presets that will be used to create an adaptive bitrate playlist.
  hls_64k_audio_preset_id = "1351620000001-200071"
  hls_0400k_preset_id     = "1351620000001-200050"
  hls_0600k_preset_id     = "1351620000001-200040"
  hls_1000k_preset_id     = "1351620000001-200030"
  hls_1500k_preset_id     = "1351620000001-200020"
  hls_2000k_preset_id     = "1351620000001-200010"

  # Additional variables
  segment_duration        = "2"
  output_key_prefix       = "elastic-transcoder-samples/output/hls/"

  input = { key: input_key }
  output_key = OpenSSL::Digest::SHA256.new(input_key.encode("UTF-8")).to_s


  hls_audio = {
    key: "hlsAudio/" + output_key,
    preset_id: hls_64k_audio_preset_id,
    segment_duration: segment_duration
  }

  hls_400k = {
    key: "hls0400k/" + output_key,
    preset_id: hls_0400k_preset_id,
    segment_duration: segment_duration
  }

  hls_600k = {
    key: "hls0600k/" + output_key,
    preset_id: hls_0600k_preset_id,
    segment_duration: segment_duration
  }

  hls_1000k = {
    key: "hls1000k/" + output_key,
    preset_id: hls_1000k_preset_id,
    segment_duration: segment_duration
  }

  hls_1500k = {
    key: "hls1500k/" + output_key,
    preset_id: hls_1500k_preset_id,
    segment_duration: segment_duration
  }

  hls_2000k = {
    key: "hls2000k/" + output_key,
    preset_id: hls_2000k_preset_id,
    segment_duration: segment_duration
  }

  outputs = [hls_audio, hls_400k, hls_600k, hls_1000k, hls_1500k, hls_2000k]
  playlist = {
    name: "hls_" + output_key,
    format: "HLSv3",
    output_keys: outputs.map { |output| output[:key] }
  }

  job = transcoder_client.create_job(
    pipeline_id: pipeline_id,
    input: input,
    output_key_prefix: output_key_prefix + output_key + "/",
    outputs: outputs,
    playlists: [playlist]
  )[:job]

  puts "HLS job has been created: " + JSON.pretty_generate(job)
end
# snippet-end:[elastictranscoder.ruby.create_hls_job.import]

if __FILE__ == $0
  transcoder_client = Aws::ElasticTranscoder::Client.new
  s3_resource = Aws::S3::Resource.new
  pipeline_name = "transcoder-pipeline-#{rand(10**4)}"
  create_elastictranscoder_job(transcoder_client, s3_resource, pipeline_name)
end